R/symbolic_units.R

```diff
@@ -136,7 +137,8 @@ as.character.symbolic_units <- function(x, ...,
       str2 <- new_denominator[j]
 
       if (are_convertible(str1, str2)) {
-        value <- convert(value, str1, str2)
+        attr(value, "units") <- str1
+        units(value) <- str2
```

Comment:
I don't see the point of this, since `units<-.units` calls `convert()`; I find this even more obfuscating than it was.

Reply:
But the point is that `units<-.units` can be intercepted by `units<-.quantities`, which is able to check whether there was a conversion, and applies it to the errors (see [here](https://github.com/r-quantities/quantities/blob/master/R/units.R#L15)).

Reply:
So I may have been unclear: conversions still take place through `convert` eventually, but this call must be enclosed in something that can be intercepted and managed from the outside.

Reply:
I trust you on this one!

Reply:
Thanks!